### ISSUE TITLE
[c] Dark color alpha can be read as 0.0 when loading JSON skeleton

### ISSUE THREAD

**Issue by jabuwu** [EsotericSoftware/spine-runtimes#2263]
Hey friends. It seems I really have a bone to pick with this JSON importer. This bug is so minor that I decided against reporting it the first time I ran into it, but I ran into it again. So, I guess I'll mention it.

Basically, the binary loader always loads dark color's alpha as 1.0:
https://github.com/EsotericSoftware/spine-runtimes/blob/5232c1cb774c0ca2510a8f4484e5f6ecd675e2ec/spine-c/spine-c/src/spine/SkeletonBinary.c#L1412

But the json loader loads it as 0.0 here:

https://github.com/EsotericSoftware/spine-runtimes/blob/5232c1cb774c0ca2510a8f4484e5f6ecd675e2ec/spine-c/spine-c/src/spine/SkeletonJson.c#L1100

Poking through json files, the alpha component isn't even saved, so I don't think it should try to read it from the file, and should be hardcoded as 1.0 here.

Thanks for your time!

**Comment by badlogic** [comment]
I agree both the JSON and binary loader should create the same color, including the (unused) alpha component. I'll fix up the JSON loader as per your suggestion.

Curious though, what specific bug did you run into that relies on the dark color alpha?

**Comment by jabuwu** [comment]
I wrote a Spine renderer for [Bevy](https://github.com/bevyengine/bevy), and then for [miniquad](https://github.com/not-fl3/miniquad). I did the same process twice, months apart, but in the same order. Both times I copy pasted this line from this repo:

https://github.com/EsotericSoftware/spine-runtimes/blob/5232c1cb774c0ca2510a8f4484e5f6ecd675e2ec/spine-ts/spine-webgl/src/Shader.ts#L264

And both times I tested the coin example, binary and JSON, because it's a good test. And I noticed there was a difference in how it renders between binary and JSON. Although, this time, I forgot why, so I spent more time than necessary hunting it down. Again.

Of course, in that shader above, the `v_dark.a` is not actually the dark color loaded from the skeleton. It's changed to 1 for PMA and 0 for non-PMA. So that's the same solution I did, and this really isn't an issue, because I think the dark color alpha shouldn't be used at all.

The inconsistency just made me lose time thinking something else was wrong, particularly, in my Rust bindings.
